core/.js/src/test/scala/io/udash/bindings/inputs/InputTest.scala

```diff
@@ -162,4 +173,77 @@ class InputTest extends AsyncUdashFrontendTest {
       r2.value should be("asd")
     }
   }
+
+  "run callback on state changes" in {
+    val p = Property[String]("ABC")
+    val result = Property("")
+    val input = TextInput(p, 0 millis, result.set(_))()
+    val inputEl = input.render
+
+    inputEl.value = "ABCD"
+    inputEl.onchange(new Event("change"))
+    result.get should be("ABCD")
+
+    inputEl.value = "ABC"
+    inputEl.onchange(new Event("change"))
+
+    result.get should be("ABC")
+
+    inputEl.value = "AB"
+    inputEl.onchange(new Event("change"))
+    result.get should be("AB")
+
+    inputEl.value = "A"
+    inputEl.onchange(new Event("change"))
+    result.get should be("A")
+
+    inputEl.value = "123qweasd"
+    inputEl.onchange(new Event("change"))
+    result.get should be("123qweasd")
+
+    p.listenersCount() should be(1)
+    input.kill()
+    p.listenersCount() should be(0)
+  }
+
+  "run callback on state changes with debouncing" in {
+    val p = Property[String]("ABC")
+    val result = Property[String]("")
+    val input = TextInput(p, 20 millis, value => {
+      result.set(value)
+    })()
+    val inputEl = input.render
+
+
+    inputEl.onchange(new Event("change"))
+    inputEl.value = "ABCD"
+
+    retrying {
+      result.get should be("ABCD")
+    } flatMap { _ =>
+      inputEl.value = "ABC"
+      inputEl.onchange(new Event("change"))
+      retrying {
+        result.get should be("ABC")
+      }
+    } flatMap { _ =>
+      inputEl.value = "AB"
+      inputEl.onchange(new Event("change"))
+      retrying {
+        result.get should be("AB")
+      }
+    } flatMap { _ =>
+      inputEl.value = "A"
+      inputEl.onchange(new Event("change"))
+      retrying {
+        result.get should be("A")
+      }
+    } flatMap { _ =>
+      inputEl.value = "123qweasd"
+      inputEl.onchange(new Event("change"))
+      retrying {
+        result.get should be("123qweasd")
+      }
+    }
+  }
```

Comment:
you're missing some tests for crucial behaviours, e.g.:
- value is set to the same (this can even happen with typing + debounce)
- different event types (supported and unsupported)

Reply:
where is the test for `value set to the same value due to debouncing`?